mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala

```diff
@@ -278,7 +278,8 @@ class DenseMatrix @Since("1.3.0") (
   }
 
   override def hashCode: Int = {
-    com.google.common.base.Objects.hashCode(numRows : Integer, numCols: Integer, toArray)
+    val state = Seq(numRows, numCols, Arrays.hashCode(values), isTransposed.hashCode)
```

Comment:
Consistent with its own `==`, are you sure? I think we're pushing around the problem then, as it's not clear this hash code is consistent with whatever definition of equality it uses. That bears investigation first.